lib/db.c

```diff
@@ -13028,8 +13028,13 @@ grn_ctx_at(grn_ctx *ctx, grn_id id)
             grn_ja_unref(ctx, &iw);
           }
           if (grn_enable_reference_count) {
-            if (!vp->ptr) {
+            if (vp->ptr->header.type == GRN_TYPE
```

Comment:
We should not touch `vp->ptr->XXX` before we check whether `vp->ptr` isn't `NULL`.